psycopg3/connection.py

```diff
@@ -110,6 +121,10 @@ def autocommit(self, value: bool) -> None:
     def _set_autocommit(self, value: bool) -> None:
         # Base implementation, not thread safe
         # subclasses must call it holding a lock
+        if len(self._savepoints) > 0:
```

Comment:
The more general `transaction_status` is below, but this one was intended to more specifically point users to say "hey, you can't touch this when in a `with conn.transaction():` block" but maybe the more general message below is good enough?

Reply:
It may be good enough. I'm ok if there is an even better message, but I'm less ok with an extra check that can be avoided. `_savepoints` non empty implies `transaction_status` `INTRANS`: maybe you can check for it, and if we know, by the second test, that the user is in a condition of sin, we can check the `_savepoints` for a better message. The fast path, "the user is not a sinner", can be checked by just looking at `transaction_status`.